File: tutorial/codelab/rotate_array/rotate_array.cpp
#include <iostream>
#include <vector>
using namespace std;

void rotateArray(int arr[], int index, int k)
{
    vector<int> v(index);

    for (int i = 0; i < index; i++)
    {
        int j = (i + k) % index;
        v[j] = arr[i];
    }

    for (int m = 0; m < v.size(); m++)
    {
        cout << v[m] << " ";
    }
}
int main()
{
    int arr[6] = {5, 2, 3, 4, 8, 9};
    int n = 6;
    int k = 3;

    rotateArray(arr, n, k);
}

File: tutorial/codelab/rotate_array/rotate_array.dart
void main() {
  List<int> arr = [5, 2, 3, 4, 8, 9];
  int n = 6;
  int k = 3;

  rotateArray(arr, n, k);
}

void rotateArray(List<int> arr, int n, int k) {
  List<int> data = List.generate(n, (index) => 0);

  for (var i = 0; i < arr.length; i++) {
    int j = (i + k) % n;
    data[j] = arr[i];
  }

  print(arr);
  print(data);
}
